spec/requests/whitehall_migration_spec.rb

```diff
@@ -0,0 +1,61 @@
+RSpec.describe "Whitehall Migration" do
+  let(:whitehall_migration) { create(:whitehall_migration) }
+  let(:user) { create(:user, permissions: [User::DEBUG_PERMISSION]) }
+
+  before do
+    login_as(user)
+  end
+
+  describe "GET /whitehall-migration/:migration_id" do
+    it "returns success" do
+      get whitehall_migration_path(whitehall_migration)
+
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "returns a forbidden status for user without debug permission" do
+       login_as(create(:user))
+       get whitehall_migration_path(whitehall_migration)
+
+       expect(response).to have_http_status(:forbidden)
+       expect(response.body).to have_content("Sorry, you don't seem to have the #{User::DEBUG_PERMISSION} permission for this app")
+    end
+  end
+
+  describe "GET /whitehall-migration/:migration_id/documents" do
+    it "returns success" do
+      create(:whitehall_migration_document_import, whitehall_migration_id: whitehall_migration.id)
+      get whitehall_migration_documents_path(whitehall_migration.id)
+
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "returns a forbidden status for user without debug permission" do
+       login_as(create(:user))
+
+       create(:whitehall_migration_document_import, whitehall_migration_id: whitehall_migration.id)
+       get whitehall_migration_documents_path(whitehall_migration.id)
+
+       expect(response).to have_http_status(:forbidden)
+       expect(response.body).to have_content("Sorry, you don't seem to have the #{User::DEBUG_PERMISSION} permission for this app")
+    end
+  end
+
+  describe "GET /whitehall-migration/:migration_id/documents/:document_import_id" do
+    it "returns success" do
+      document_import = create(:whitehall_migration_document_import, whitehall_migration_id: whitehall_migration.id)
+      get whitehall_migration_document_path(whitehall_migration.id, document_import.id)
+
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "returns a forbidden status for user without debug permission" do
+       login_as(create(:user))
+       document_import = create(:whitehall_migration_document_import, whitehall_migration_id: whitehall_migration.id)
+       get whitehall_migration_document_path(whitehall_migration.id, document_import.id)
+
+       expect(response).to have_http_status(:forbidden)
+       expect(response.body).to have_content("Sorry, you don't seem to have the #{User::DEBUG_PERMISSION} permission for this app")
+    end
+  end
```

Comment:
Nice job putting together these request tests. 

I think we can use an existing helper to handle the permission cases more concisely: https://github.com/alphagov/content-publisher/blob/6506cabb5109eabbc14a4048340be27891044823/spec/requests/file_attachments_spec.rb#L17-L26

You could do something like: 

```
  it_behaves_like "requests that return status",
                  "when a user without debug permissions looks at a whitehall migration",
                  status: :forbidden,
                  routes: { whitehall_migration: %i[get],
                            whitehall_migration_documents: %i[get] } do
    before { login_as(create(:user)) }
    let(:route_params) { [whitehall_migration] }
  end
  
  it_behaves_like "requests that return status",
                  "when a user without debug permissions looks at a whitehall migration document",
                  status: :forbidden,
                  routes: { whitehall_migration_document: %i[get] } do
    before { login_as(create(:user)) }
    let(:route_params) do
      document = create(:whitehall_migration_document_import, whitehall_migration: whitehall_migration)
      [whitehall_migration, document]
    end
  end  
```

Reply:
👍